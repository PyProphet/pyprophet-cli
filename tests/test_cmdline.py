# encoding: utf-8
from __future__ import print_function
import pdb
import subprocess
import pytest
import os
import shutil
import collections
import tempfile

import pandas

pandas.options.display.precision=6
pandas.options.display.width=200


@pytest.fixture(scope="session")
def test_data_folder():
    here = os.path.dirname(os.path.abspath(__file__))
    return os.path.join(here, "data")


@pytest.fixture(scope="session", params=[1, 3])
def setup(test_data_folder, request):
    num_files = request.param
    tmpdir = tempfile.mkdtemp()
    data_folder = os.path.join(tmpdir, "data_folder")
    os.makedirs(data_folder)
    working_folder = os.path.join(tmpdir, "working_folder")
    for i in range(num_files):
        shutil.copy(os.path.join(test_data_folder, "test_data.txt"), os.path.join(data_folder, "data_%d.txt" % i))

    Setup = collections.namedtuple("Setup", "data_folder working_folder number_input_files")
    return Setup(data_folder, working_folder, num_files)


def test_check(setup):
    cmd = ("pyprophet-brutus check --data-folder %s" % setup.data_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0


def test_subsample(setup):
    cmd = ("pyprophet-brutus subsample --job-number 1 --job-count 1 "
           "--sample-factor 0.1 "
           "--random-seed 43 "
           "--ignore-invalid-scores "
           "--data-filename-pattern '*.txt' "
           "--data-folder %s --working-folder %s") % (setup.data_folder, setup.working_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0
    files = os.listdir(setup.working_folder)
    assert len(files) == setup.number_input_files + 1

    files.sort()
    subsamples = pandas.read_csv(os.path.join(setup.working_folder, files[-1]), sep="\t")
    assert subsamples.shape == (934, 21)

    cmd = ("pyprophet-brutus subsample --job-number 1 --job-count 2 "
           "--sample-factor 0.1 "
           "--random-seed 43 "
           "--ignore-invalid-scores "
           "--data-filename-pattern '*.txt' "
           "--data-folder %s --working-folder %s "
           "--local-folder %s") % (setup.data_folder, setup.working_folder, tempfile.mkdtemp())
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0
    files = os.listdir(setup.working_folder)
    assert len(files) == 2 if setup.number_input_files == 1 else 3

    files.sort()
    subsamples = pandas.read_csv(os.path.join(setup.working_folder, files[-1]), sep="\t")
    assert subsamples.shape == (934, 21)

    cmd = ("pyprophet-brutus subsample --job-number 1 --job-count 2 "
           "--sample-factor 0.1 "
           "--random-seed 43 "
           "--ignore-invalid-scores "
           "--data-filename-pattern '*.txt' "
           "--data-folder %s --working-folder %s") % (setup.data_folder, setup.working_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0
    files = os.listdir(setup.working_folder)
    assert len(files) == 2 if setup.number_input_files == 1 else 3

    files.sort()

    subsamples = pandas.read_csv(os.path.join(setup.working_folder, files[-1]), sep="\t")
    assert subsamples.shape == (934, 21)


def test_learn(setup, regtest):
    cmd = ("pyprophet-brutus learn --random-seed 43 --working-folder %s" % setup.working_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0

    files = sorted(os.listdir(setup.working_folder))

    for i, file_ in enumerate(files):
        path = os.path.join(setup.working_folder, file_)
        print(i, "%7d" % os.stat(path).st_size, file_, file=regtest)

    for name in ("weights.txt", "sum_stat_subsampled.txt"):
        df = pandas.read_csv(os.path.join(setup.working_folder, name), header=None, sep="\t")
        print(df, file=regtest)
        # print(open(os.path.join(setup.working_folder, name), "r").read(), file=regtest)


def test_score(setup, regtest):
    cmd = ("pyprophet-brutus apply_weights --job-number 1 --job-count 1 "
           "--data-filename-pattern '*.txt' "
           "--data-folder %s --working-folder %s") % (setup.data_folder, setup.working_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0
