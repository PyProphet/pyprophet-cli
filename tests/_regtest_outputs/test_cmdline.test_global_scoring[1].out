 0      20 work_folder/extra_group_columns.txt
 1    6460 work_folder/full_stat_subsampled.txt
 2      17 work_folder/invalid_columns.txt
 3     392 work_folder/score_columns.txt
 4 1049943 work_folder/subsampled_data_0.txt
 5     706 work_folder/sum_stat_subsampled.txt
 6     433 work_folder/weights.txt
            0
0    0.204622
1   -0.113499
2   -3.198922
3    1.234337
4    2.267385
5   -2.008769
6   -0.575658
7  -11.174708
8    0.115012
9    0.009342
10  -0.006851
11  -8.766545
12   0.303427
13  -0.559181
14   4.110611
15   2.144420
16   0.117337   qvalue    svalue        pvalue   TP  FP         TN          FN           FDR      sens    cutoff
0    0.00  0.015385  4.458236e-10    3   0  12.061856  179.938144  1.910673e-09  0.016399  6.323315
1    0.01  0.986590  1.367148e-01  180   2  10.061856    2.938144  9.053957e-03  0.983939  1.492660
2    0.02  1.000000  3.115527e-01  183   4   8.061856   -0.061856  2.010017e-02  1.000000  0.600435
3    0.05  1.000000  8.368489e-01  184  10   2.061856   -1.061856  5.197819e-02  1.000000 -0.246332
4    0.10  1.000000  9.148045e-01  184  11   1.061856   -1.061856  5.658584e-02  1.000000 -1.180629
5    0.20       NaN           NaN  NaN NaN        NaN         NaN           NaN       NaN       NaN
6    0.30       NaN           NaN  NaN NaN        NaN         NaN           NaN       NaN       NaN
7    0.40       NaN           NaN  NaN NaN        NaN         NaN           NaN       NaN       NaN
8    0.50       NaN           NaN  NaN NaN        NaN         NaN           NaN       NaN       NaN 0   89146 work_folder/data_0_score_data.npz
 1 2162728 work_folder/data_0_top_score_data.hdf
 2      20 work_folder/extra_group_columns.txt
 3    6460 work_folder/full_stat_subsampled.txt
 4      17 work_folder/invalid_columns.txt
 5     392 work_folder/score_columns.txt
 6 1049943 work_folder/subsampled_data_0.txt
 7     706 work_folder/sum_stat_subsampled.txt
 8     433 work_folder/weights.txt
 0 2825717 result_folder_1/data_0_scored.txt
 1   29728 result_folder_1/report.pdf
 2   29728 result_folder_1/report_grouped_by_transition_group_id.pdf
 3    1165 result_folder_1/summary_stats.txt
 4    1165 result_folder_1/summary_stats_grouped_by_transition_group_id.txt
 0   89146 work_folder/data_0_score_data.npz
 1 2162728 work_folder/data_0_top_score_data.hdf
 2      20 work_folder/extra_group_columns.txt
 3    6460 work_folder/full_stat_subsampled.txt
 4      17 work_folder/invalid_columns.txt
 5     392 work_folder/score_columns.txt
 6 1049943 work_folder/subsampled_data_0.txt
 7     706 work_folder/sum_stat_subsampled.txt
 8     433 work_folder/weights.txt

num_null   : 25.00
num_total  : 387.00
stats shape: (387, 10)
mean top target scores: 4.367875
sdev top target scores: 1.518134
mean top decoy  scores: -0.000000
sdev top decoy  scores: 1.000000

   qvalue    svalue    pvalue   TP  FP            TN            FN       FDR      sens    cutoff
0    0.00  0.872915  0.000186  316   0  2.500000e+01  4.600000e+01  0.000212  0.872928  3.559004
1    0.01  0.998093  0.147615  361   4  2.100000e+01  1.000000e+00  0.010111  0.997238  1.046715
2    0.02  1.000000  0.282893  361   7  1.800000e+01  1.000000e+00  0.019218  0.997238  0.574269
3    0.05  1.000000  0.795019  363  20  5.000000e+00 -1.000000e+00  0.051894  1.000000 -0.823962
4    0.10  1.000000  0.982478  362  25 -3.552714e-15  3.552714e-15  0.063468  1.000000 -2.107861
5    0.20       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
6    0.30       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
7    0.40       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
8    0.50       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN

num_null   : 25.00
num_total  : 387.00
stats shape: (387, 10)
mean top target scores: 4.367875
sdev top target scores: 1.518134
mean top decoy  scores: -0.000000
sdev top decoy  scores: 1.000000

   qvalue    svalue    pvalue   TP  FP            TN            FN       FDR      sens    cutoff
0    0.00  0.872915  0.000186  316   0  2.500000e+01  4.600000e+01  0.000212  0.872928  3.559004
1    0.01  0.998093  0.147615  361   4  2.100000e+01  1.000000e+00  0.010111  0.997238  1.046715
2    0.02  1.000000  0.282893  361   7  1.800000e+01  1.000000e+00  0.019218  0.997238  0.574269
3    0.05  1.000000  0.795019  363  20  5.000000e+00 -1.000000e+00  0.051894  1.000000 -0.823962
4    0.10  1.000000  0.982478  362  25 -3.552714e-15  3.552714e-15  0.063468  1.000000 -2.107861
5    0.20       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
6    0.30       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
7    0.40       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
8    0.50       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN

transition_group_id	run_id	decoy	main_var_xx_swath_prelim_score	var_bseries_score	var_elution_model_fit_score	var_intensity_score	var_isotope_correlation_score	var_isotope_overlap_score	var_library_corr	var_library_rmsd	var_log_sn_score	var_massdev_score	var_massdev_score_weighted	var_norm_rt_score	var_xcorr_coelution	var_xcorr_coelution_weighted	var_xcorr_shape	var_xcorr_shape_weighted	var_yseries_score	var_faulty_score	peak_group_rank	d_score	m_score	p_value	transition_group_id_m_score	transition_group_id_p_value

459_run0	0	0	5.27893972397	5.0	0.993098139763	0.930480718613	0.998724222183	0.0	0.912406027317	0.0398625209928	5.01656723022	8.83918857574	3.88453722	0.00229200161994	0.0	0.0	0.960350096226	0.970979332924	6.0		1	5.00816392899	0.000211860926356	2.74758349406e-07	0.000211860926356	2.74758349406e-07

